src/pages/ReportDetailsPage.tsx

```diff
@@ -263,6 +264,25 @@ function ReportDetailsPage({policies, report, session, personalDetails}: ReportD
     const shouldShowWriteCapability = !isMoneyRequestReport;
     const shouldShowMenuItem = shouldShowNotificationPref || shouldShowWriteCapability || (!!report?.visibility && report.chatType !== CONST.REPORT.CHAT_TYPE.INVOICE);
 
+    const isPayer = ReportUtils.isPayer(session, moneyRequestReport);
+    const isSettled = ReportUtils.isSettled(moneyRequestReport?.reportID ?? '-1');
+
+    const shouldShowCancelPaymentButton = caseID === CASES.MONEY_REPORT && isPayer && isSettled && ReportUtils.isExpenseReport(moneyRequestReport);
+    const [chatReport] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${moneyRequestReport?.chatReportID ?? '-1'}`);
+
+    const cancelPayment = useCallback(() => {
+        if (!chatReport) {
+            return;
+        }
+
+        if (!moneyRequestReport) {
+            return;
+        }
```

Comment:
Why are we adding this extra condition? I don't see it being used in `MoneyReportHeader` component.

Reply:
This sort of check has to be added here, since for `ReportDetailsPage`, which combines `MoneyReportHeader`, `MoneyRequestHeader` and `HeaderView` logic after Details Revamp, `moneyRequestReport` might be undefined in other cases, unlike in `MoneyReportHeader`, which had it as a required argument passed to the component.

I have moved this logic into the `IOU.cancelPayment();` function and changed it to support `OnyxEntry<OnyxTypes.Report>`. This unifies the cancel function with other IOU functions such as `IOU.unapproveExpenseReport();` and should be a good solution here.